src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -68,35 +64,19 @@ def run_python_test(test_target, pytest, python_setup, source_root_config, subpr
   # https://pytest.org/en/latest/goodpractices.html#test-discovery. In addition to a performance
   # optimization, this ensures that any transitive sources, such as a test project file named
   # test_fail.py, do not unintentionally end up being run as tests.
-  source_roots = source_root_config.get_source_roots()
-  test_target_sources_file_names = []
-  for source_target_filename in test_target.sources.files_relative_to_buildroot:
-    source_root = source_roots.find_by_path(source_target_filename)
-    test_target_sources_file_names.append(
-      strip_prefix(source_target_filename, prefix=f"{source_root.path}/")
+
+  hydrated_target = HydratedTarget(address="", adaptor=test_target, dependencies=())
```

Comment:
Since there is a rule to map `BuildFileAddress` to `HydratedTarget`, instead of creating one here, can you just change the Get below to be `Get(SourceRootStrippedSources, BuildFileAddress, test_target.address)`. I could be off by s/BuildFileAddress/Address/ in which case you'll need `Get(SourceRootStrippedSources, Address, test_target.address.to_address)`.